### ISSUE TITLE
Looping checks rather than fixed delays

### ISSUE THREAD

**Issue by AlexanderPico** [cytoscape/RCy3#101]
Per Barry's suggestion, we could check for certain events or changes to complete before proceeding instead of relying on the hard-coded delays which are not optimized per use case.

Might borrow ideas from here to construct the looping checks:
http://bioconductor.org/developers/how-to/web-query/

**Comment by AlexanderPico** [comment]
Might also include setters for the delay values, like so:
```
def set_catchup_filter_secs(delay_secs):
    CATCHUP_FILTER_SECS = delay_secs

def set_model_propagation_secs(delay_secs):
    MODEL_PROPAGATION_SECS = delay_secs

def set_catchup_network_secs(delay_secs):
    CATCHUP_NETWORK_SECS = delay_secs

def set_ndex_delay_secs(delay_secs):
    NDEX_DELAY_SECS = delay_secs
```

**Comment by bdemchak** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Looping checks rather than fixed delays #101](https://trello.com/c/aIKOrJvB/23-looping-checks-rather-than-fixed-delays-101)

**Comment by AlexanderPico** [comment]
Removed NDEx delays (like in py4cy) and added setter methods in RCy3.R using custom package environment, `RCy3env`.
